runtime/doc/treesitter.txt

```diff
@@ -523,6 +530,9 @@ associated with patterns:
     node's entire text should be re-parsed, including the text of its child
     nodes. By default, child nodes' text will be excluded from the injected
     document.
+    • `injection.self` - indicates that the node's text should be parsed with
```

Comment:
Duplicated

Reply:
It's somewhat intentional - one's for `@injection.self`, the other `(#set! injection.self)`

Reply:
What's the rationale for the duplication upstream here (and not for self.parent)? Do we need to support (officially, documented) both?

Reply:
I made `injection.self` both for consistency w/ `injection.language`

But since we know the language in the self/parent instances both times, I could remove the set directive version (or the capture), or add the directive for parent

Reply:
I think it's better to support only one way of doing things, unless there's a good use case for both.

I personally don't have much of a preference which (probably the directive is more consisy); the main thing is to be consistent with upstream in order to be able to share queries more easily.
(Of course, that means if upstream supports both, we need to as well -- grudgingly.)

@lewis6991 ?

Reply:
I can just remove one choice from upstream's self version and keep it consistent here, it was my (not-so-well-made) decision after all :grin: 

Question is which - I'm really indifferent

Reply:
Yes, that would be good (to remove one upstream to have the same single one for each). It's probably easier to discuss upstream which?

Reply:
Yeah I can ask ahlinc, though it was pretty much my choice to add both

Reply:
If you really can't decide, my vote would be for keeping the directives.

Reply:
Yeah still kinda mulling it over - I think directives is more in-line w/ the upstream style

Reply:
Yep, that's exactly why I suggested it. :)

Reply:
https://github.com/tree-sitter/tree-sitter/pull/2550

Reply:
I agree with only directives, as captures here may be confusing in usage. Should it be placed where `injection.language` should be, or should it be similar to nvim-treesitter's old `@{lang}`, replacing `injection.content` ? Both are reasonable.

Reply:
I think I see why Max structured it as such for injections specifically - captures are only meant to be used if the node's text is relevant to the injection, else you're just wasting space in the fields of the HighlightConfiguration (hence why you read the string content of the directive's key dynamically)

Reply:
Updated to using directives only